lib/models/api_calls.dart

```diff
@@ -0,0 +1,92 @@
+  ///Url to get the intial page of the list of devices
+  static final _deviceListUrl =
+      'https://www.googleapis.com/admin/directory/v1/customer/my_customer/devices/chromeos?projection=BASIC&maxResults=15';
+
+  ///Url to get the next page of the list of devices
+  static final _deviceListWithTokenUrl =
+      'https://www.googleapis.com/admin/directory/v1/customer/my_customer/devices/chromeos?projection=BASIC&maxResults=15&pageToken=';
+
+  ///Url to get detailed information of a device
+  static final _detailedDeviceUrl =
+      'https://www.googleapis.com/admin/directory/v1/customer/my_customer/devices/chromeos/';
+
+  /// Calls the Directory API to get a list of devices with max results of 15.
+  ///
+  /// * [client] is the http client that will do the https requests.
+  /// * [authToken] is the authentication token for API to accept the requests.
+  /// * [nextPageToken]  is the next page of the devices list that should
+  /// retreive, if it is null the function will request the first page.
+  ///
+  /// This method returns a [AccountDevices] object.
+  static getDeviceList(
+      http.Client client, String authToken, String nextPageToken) async {
+    var response;
+    authToken = 'Bearer $authToken';
+    if (nextPageToken == null) {
+      response = await client.get(_deviceListUrl,
+          headers: {HttpHeaders.authorizationHeader: authToken});
+    } else {
+      response = await client.get(_deviceListWithTokenUrl + nextPageToken,
+          headers: {HttpHeaders.authorizationHeader: authToken});
+    }
+    // If conection is succesful return a AccountDevices object by parsing the
+    // response body
+    if (response.statusCode == 200) {
+      return AccountDevices.fromJson(json.decode(response.body));
+    } else {
+      // If that call was not successful, throw an error.
+      throw Exception('${response.statusCode} ${response.body}');
+    }
+  }
+
+  /// Calls the Directory API to get the detailed information of a device.
+  ///
+  /// * [client] is the http client that will do the https requests.
+  /// * [authToken] is the authentication token for API to accept the requests.
+  /// * [deviceId] is the ID of the device which its detailed information
+  /// will be retrieved.
+  ///
+  /// This method returns a [DetailedDevice] object.
+  static dynamic getDetailedDevice(
+      http.Client client, String authToken, String deviceId) async {
+    authToken = 'Bearer $authToken';
+    final response = await client.get(_detailedDeviceUrl + deviceId,
+        headers: {HttpHeaders.authorizationHeader: authToken});
+    // If conection is succesful return a DetailedDevice object by parsing the
+    // response body
+    if (response.statusCode == 200) {
+      return DetailedDevice.fromJson(json.decode(response.body));
+    } else {
+      // If that call was not successful, throw an error.
+      throw Exception('${response.statusCode} ${response.body}');
+    }
+  }
+
+  // TODO Remote commands ...
```

Comment:
Please add the issue and put down the issue number or link to the issue

Reply:
#38